CosmosDb.GraphAPI.Recommender/DataProvider.cs

```diff
@@ -0,0 +1,472 @@
+                }
+            }
+        }
+
+        // Testing
+
+        public static void FindCommonProductsBetweenPeople(List<Person> people)
+        {
+            // int - product id
+            // List<int> - people id
+            var commonProducts = new Dictionary<int, List<int>>();
+
+            foreach (var person in people)
+            {
+                foreach (var product in person.ProductIds)
+                {
+                    if (!commonProducts.ContainsKey(product))
+                    {
+                        commonProducts.Add(product, new List<int>());
+                    }
+
+                    commonProducts[product].Add(person.Id);
+                }
+            }
+
+            //int GetCommonCount(int[] ar1, int[] ar2)
+            //{
+            //    int result = 0;
+            //    for (int i = 0; i < ar1.Length; ++i)
+            //    {
+            //        for (int j = 0; j < ar2.Length; ++j)
+            //        {
+            //            if (ar1[i] == ar2[j])
+            //            {
+            //                ++result;
+            //            }
+            //        }
+            //    }
+
+            //    return result;
+            //}
+
+            ////Сколько общего между каждым пользователем
+
+            //int[,] ma = new int[people.Count, people.Count];
+
+            //for (int i = 0; i < people.Count; ++i)
+            //{
+            //    for (int j = 0; j < people.Count; ++j)
+            //    {
+            //        ma[i, j] = GetCommonCount(people[i].ProductIds, people[j].ProductIds);
+            //    }
+            //}
+
+            //using (var sw = new StreamWriter(@"C:\Users\Ruslan_Bondar\Projects\AzureSamples\CosmosDb.GraphAPI.Recommender\Data\CommonTable.csv"))
+            //{
+            //    for (int i = 0; i < people.Count; ++i)
+            //    {
+            //        for (int j = 0; j < people.Count; ++j)
+            //        {
+            //            sw.Write(ma[i, j] + ",");
+            //        }
+            //        sw.WriteLine();
+            //    }
+            //}
+
+            //Console.WriteLine("Product : People who have this product");
+
+            using (var sw = new StreamWriter(@"C:\Users\Ruslan_Bondar\Projects\AzureSamples\CosmosDb.GraphAPI.Recommender\Data\AdjencyList.csv"))
+            {
+                sw.WriteLine("product id, people ids who have");
+                foreach (var productPeoplePair in commonProducts)
+                {
+                    sw.Write(productPeoplePair.Key + ",");
+
+                    for (int i = 0; i < productPeoplePair.Value.Count; ++i)
+                    {
+                        sw.Write(productPeoplePair.Value[i] + ",");
+                    }
+
+                    sw.WriteLine();
+                }
+            }
+        }
+
+        // Read stock data
+
+        private void ReadStockBrandsProductsData()
```

Comment:
Is it related to data generator? If not, then probably let's exclude it from here and place somewhere at reader implementation